cogs/job_offers.py

```diff
@@ -0,0 +1,152 @@
+
+        embed = disnake.Embed(title="Stellenangebote der Uni",
+                              description=f"Ich habe folgende Stellenangebote der Fakultät {faculty} gefunden:")
+        if offers := self.joboffers.get(faculty):
+            for offer in offers:
+                offer = offers[offer]
+                descr = f"{offer['info']}\nDeadline: {offer['deadline']}\n{offer['link']}"
+                embed.add_field(name=offer['title'], value=descr, inline=False)
+
+        await interaction.response.send_message(embed=embed)
+
+
+    async def post_new_jobs(self, jobs):
+        embed = disnake.Embed(title="Neue Stellenangebote der Uni",
+                              description=f"Ich habe folgende neue Stellenangebote der Fakultät {STD_FAK} gefunden:")
+        for job in jobs:
+            descr = f"{job['info']}\nDeadline: {job['deadline']}\n{job['link']}"
+            embed.add_field(name=job['title'], value=descr, inline=False)
+
+        joboffers_channel = await self.bot.fetch_channel(self.joboffers_channel_id)
+        await joboffers_channel.send(embed=embed)
+
+
+    async def fetch_joboffers(self):
+        sess = aiohttp.ClientSession()
+        req = await sess.get(JOBS_URL)
+        text = await req.read()
+        await sess.close()
+
+        soup = BeautifulSoup(text, "html.parser")
+        list = soup.findAll("li")
+
+        # alte Liste sichern zum Abgleich
+        old_joboffers = deepcopy(self.joboffers)
+
+        for job in list:
+            detail_string = job.text.strip()
+            if "Studentische Hilfskraft" in detail_string:
+                id = detail_string[detail_string.index('(')+12:detail_string.index(')')]
+                title = detail_string[:detail_string.index(')')+1]
+                info = detail_string[detail_string.index(')')+1:detail_string.index('[')]
+                deadline = detail_string[detail_string.index('[')+1:detail_string.index(']')]
+                link = job.find('a')['href']
+
+                # Umlaute aufräumen
+                info = info.replace("Ã¤","ä")
+                info = info.replace("Ã¼", "ü")
+
+                faks = ["other", "wiwi", "mi", "ksw", "psy", "rewi"]
+
+                fak_id = int(id[0]) # Kennziffer 1=wiwi, 2=mi, 3=ksw, 4=psy, 5=rewi, alle anderen=other
+                if fak_id in range(1,6):
+                    fak = faks[fak_id]
+                else:
+                    fak = faks[0]
+
+                if not self.joboffers.get(fak):
+                    self.joboffers[fak] = {}
+                self.joboffers[fak][id] = {'title': title, 'info': info, 'deadline': deadline, 'link': link}
+        self.save_joboffers()
+        await self.check_for_new_jobs(old_joboffers)
+
+    async def check_for_new_jobs(self, old_joboffers):
+        new_jobs = []
+        if fak := self.joboffers.get(STD_FAK):
+            if fak_old := old_joboffers.get(STD_FAK):
+                for offer_id in fak:
+                    offer = fak[offer_id]
```

Comment:
Eines der `if` kann auch hier durch die for Schleife ersetzt werden. Also beispielsweise
```python
if fak_old := old_joboffers.get(STD_FAK):
    for offer_id, offer_data in self.joboffers.get(STD_FAK).items():
```